wfcGen.py

```diff
@@ -2,83 +2,78 @@
+        # [left, top, right, bottom]uwu
+        # u will newer need that
 
-    for y in range(lcy):
-        for x in range(lcx):
-            flag = -1
-            for key in cells.keys():
-                if cropped[ins][y][x] == cells[key]["img"]:
-                    flag = key
+        for y in range(lcy):
+            for x in range(lcx):
+                cellId = -1
+                for CellIdKey in cells.keys():
+                    if cropped[ins][y][x] == cells[CellIdKey]["img"]:
+                        cellId = CellIdKey
 
-            if flag != -1:
-                cells[flag]["nbg"][0].append(cropped[ins][y][x-1-(res-1)*int(cropShift)])
-                cells[flag]["nbg"][1].append(cropped[ins][y-1-(res-1)*int(cropShift)][x])
-                cells[flag]["nbg"][2].append(cropped[ins][y][(x+1+(res-1)*int(cropShift))%lcx])
-                cells[flag]["nbg"][3].append(cropped[ins][(y+1+(res-1)*int(cropShift))%lcy][x])
-            else:
-                cells[str(len(cells.keys()))] = {
-                    "nbg" : {
-                        "0" : [cropped[ins][y][x-1-(res-1)*int(cropShift)]],
-                        "1" : [cropped[ins][y-1-(res-1)*int(cropShift)][x]],
-                        "2" : [cropped[ins][y][(x+1+(res-1)*int(cropShift))%lcx]],
-                        "3" : [cropped[ins][(y+1+(res-1)*int(cropShift))%lcy][x]],
-                    }, 
-                    "img" : cropped[ins][y][x]
-                }
-
-    for y in range(lcy):
-        for x in range(lcx):
-            for key in cells.keys():
-                if cells[key]["img"] == cropped[ins][y][x]:
-                    Gkey = key
-            for key in cells.keys():
-                for key2 in cells[key]["nbg"].keys():
-                    if key2 != "img":
-                        for i in range(len(cells[key]["nbg"][key2])):
-                            if cells[key]["nbg"][key2][i] == cropped[ins][y][x]:
-                                cells[key]["nbg"][key2][i] = Gkey
-
-    print(cells)
+                if cellId != -1:
+                    cells[cellId]["nbg"]["left"].append(cropped[ins][y][x-1-(tileSize-1)*int(cropShift)])
+                    cells[cellId]["nbg"]["top"].append(cropped[ins][y-1-(tileSize-1)*int(cropShift)][x])
+                    cells[cellId]["nbg"]["right"].append(cropped[ins][y][(x+1+(tileSize-1)*int(cropShift))%lcx])
+                    cells[cellId]["nbg"]["bottom"].append(cropped[ins][(y+1+(tileSize-1)*int(cropShift))%lcy][x])
+                else:
+                    cells[len(cells.keys())] = {
+                        "nbg" : {
+                            "left" : [cropped[ins][y][x-1-(tileSize-1)*int(cropShift)]],
+                            "top" : [cropped[ins][y-1-(tileSize-1)*int(cropShift)][x]],
+                            "right" : [cropped[ins][y][(x+1+(tileSize-1)*int(cropShift))%lcx]],
+                            "bottom" : [cropped[ins][(y+1+(tileSize-1)*int(cropShift))%lcy][x]],

```

Comment:
These expressions are still hard to read. You can definitely improve them by extracting some subexpressions (e.g., `(tileSize - 1) * int(cropShift)`? Anything else?) into separate variables with names. This will both give more insight into what's going on (because there is a named variable), and simplify the index expressions